### PR TITLE
add snapshot to CI

### PR DESCRIPTION
**Pull request by snreynolds**
## Related Issue
Which issue does this pull request resolve?

We should be running forge snapshots on all PRs locally & on CI. There was an issue with getting the fuzz snapshots locally to match the CI. Hopefully we can resolve it in this PR.

## Description of changes

### CODE REVIEW

**Comment by snreynolds** [comment]
trying to fix #296

**Comment by marktoda** [comment]
tl;dr had to set the fuzz seed _in the environment_ because forge snapshot does not respect it from foundry.toml currently (issue below). So now we should use the `justfile` to run with the proper seed i.e. `just snapshots`

Reference to foundry issue https://github.com/foundry-rs/foundry/issues/6179

File: foundry.toml [hunk 1]
```diff
@@ -8,7 +8,11 @@ ffi = true
 fs_permissions = [{ access = "read-write", path = ".forge-snapshots/"}, { access = "read", path = "./foundry-out"}]
 evm_version = 'paris'
 
-[profile.ci]
-fuzz_runs = 100000
+[profile.default.fuzz]
+runs = 100
```
Comment (snreynolds) [inline foundry.toml:12]: why are the runs different in default vs ci?

Comment (marktoda) [reply foundry.toml:12]: So it runs faster locally! Lots of runs can make it quite slow esp with lot of fuzz tests

Comment (marktoda) [reply foundry.toml:12]: So it runs faster locally! Lots of runs can make it quite slow esp with lot of fuzz tests

**Comment by snreynolds** [comment]
closing this in favor of this https://github.com/Uniswap/v4-core/pull/401 just squashes these, rebases main, and adds a tolerance

**Pull request closed without merging**